core/domain/state_domain.py

```diff
@@ -1471,11 +1471,11 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                 'Expected solicit_answer_details to be a boolean, ' +
                 'received %s' % self.solicit_answer_details)
         if self.solicit_answer_details:
-            if self.interaction.id in feconf.CANNOT_SOLICIT_ANSWER_DETAILS:
+            if self.interaction.id in (
+                    feconf.INTERACTION_IDS_WITHOUT_ANSWER_DETAILS):
                 raise utils.ValidationError(
-                    'This state cannont have true instance of ' +
-                    'solicit_answer_details, because the interaction is ' +
-                    'of type %s.' % self.interaction.id)
+                    'The %s interaction does not ' % (self.interaction.id) +
```

Comment:
Prefer: 'The %s interaction does not support soliciiting answer details from learners.' % self.interaction.id

I.e., do the interpolation at the very end.

Reply:
Actually I earlier did the same, but the ```% self.interaction.id``` moves to the next line, so its gives the error as
```
TypeError: not all arguments converted during string formatting

```

Reply:
Maybe because you are using `+` (concat).

Reply:
Yeah Sandeep I have to use ```+``` as it moves to next line.

Reply:
![Screenshot from 2019-06-13 14-27-32](https://user-images.githubusercontent.com/30555544/59418765-39e85500-8de7-11e9-82aa-6e12beecc205.png)

Reply:
In Python, you don't. Try removing the "+".

Reply:
Done